tools/lkl/lib/virtio_net_tap.c

```diff
@@ -122,5 +196,24 @@ struct lkl_netdev *lkl_netdev_tap_create(const char *ifname)
 		return NULL;
 	}
 
+	#ifdef CONFIG_EVENTFD
```

Comment:
Basically I don't think the code in virtio_net_tap should care about anything of hijack. The hijack lib should make sure it works.

One possible solution is making lkl_netdev_tap_create accept an function pointer which by default is eventfd. This at least make it possible to have eventfd and free up mem safely.

More generally, maybe we should have a struct lkl_tap_host_ops, which defines all the host ops needed by tap. 
struct lkl_tap_host_ops {
  int (*eventfd)(unsigned int, int);
}

It's exposed by virtio_net_tap.h and modified by hijack lib. Dont know if the virtio tap is the only reason we have the hacky 500 fd limit. But this also can help to solve it.